scripts/modules/elastic_stack.py

```diff
@@ -587,6 +587,46 @@ def render_tee(self):
         return {self.name(): content}
 
 
+class PackageRegistry(StackService, Service):
+
+    SERVICE_PORT = "8080"
+
+    docker_path = "package-registry"
+    docker_name = "distribution"
+
+    def __init__(self, **options):
+        super(PackageRegistry, self).__init__(**options)
+        self.distribution = options.get("package_registry_distribution", "snapshot")
+        self.apm_package = options.get("apm_package")
+        if not self.at_least_version("7.10"):
+            raise Exception("Package registry only supported for 7.70+")
+        self.environment = {}
+
+    def _content(self):
+        content = dict(
+            image="/".join((self.docker_registry, self.docker_path, self.docker_name)) + ":" + self.distribution,
+            environment=self.environment,
+            healthcheck=curl_healthcheck(self.SERVICE_PORT, path="/", interval="5s", retries=10),
+            ports=[self.publish_port(self.SERVICE_PORT)]
+        )
+        if self.apm_package:
+            content.update(volumes=[self.apm_package + ":/packages/" + self.distribution + "/apm"])
+        return content
+
+    @classmethod
+    def add_arguments(cls, parser):
+        super(PackageRegistry, cls).add_arguments(parser)
+        parser.add_argument(
+            "--package-registry-distribution",
+            default="snapshot",
+            help="Package registry distribution: snapshot | staging | production"
+        )
+        parser.add_argument(
+            "--apm-package",
```

Comment:
In keeping with the naming of other options:

```suggestion
            "--package-registry-apm-path",
```